spring-integration-core/src/main/java/org/springframework/integration/util/CheckedFunction.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.util;
+
+import java.util.function.Function;
+
+/**
+ * A Function-like interface which allows throwing Error.
+ *
+* @param <T> the input type.
+* @param <R> the output type.
+ *
+ * @author Artem Bilan
+ *
+ * @since 6.1
+ */
+@FunctionalInterface
+public interface CheckedFunction<T, R> {
+
+	R apply(T t) throws Throwable;
+
+	default Function<T, R> unchecked() {
+		return t1 -> {
+			try {
+				return apply(t1);
+			}
+			catch (Throwable t) {
+				return sneakyThrow(t);
+			}
+		};
+	}
```

Comment:
I am not really comfortable with institutionalizing this JVM abuse. Is there no alternative?

Reply:
OK. Will it be better if there is no that `unchecked()` then?
Something like plain:
```
@FunctionalInterface
public interface CheckedFunction<T, R> {

	R apply(T t) throws Throwable;

}
```
?

Reply:
Right; that's the bit I was objecting to. You probebly need a `// NOSONAR` here too.